src/github/api.rs

```diff
@@ -0,0 +1,102 @@
+use std::collections::HashMap;
+
+use anyhow::Context;
+use octocrab::models::issues::Comment;
+use octocrab::models::{AppId, InstallationRepositories};
+use octocrab::Octocrab;
+use secrecy::{ExposeSecret, SecretVec};
+
+use crate::github::GitHubRepositoryKey;
+
+/// Provides access to managed GitHub repositories.
+#[derive(Debug)]
+pub struct RepositoryAccess {
+    repositories: HashMap<GitHubRepositoryKey, Repository>,
+}
+
+impl RepositoryAccess {
+    /// Loads repositories managed by the Bors GitHub app with the given ID.
+    pub async fn load_repositories(
+        app_id: AppId,
+        private_key: SecretVec<u8>,
+    ) -> anyhow::Result<RepositoryAccess> {
+        let key = jsonwebtoken::EncodingKey::from_rsa_pem(private_key.expose_secret().as_ref())
+            .context("Cannot encode private key")?;
+
+        let octocrab = Octocrab::builder()
+            .app(app_id, key)
+            .build()
+            .context("Cannot create octocrab builder")?;
+
+        let installations = octocrab
+            .apps()
+            .installations()
+            .send()
+            .await
+            .context("Cannot load app installations")?;
+
+        let mut repositories = HashMap::default();
+        for installation in installations {
+            if let Some(ref repositories_url) = installation.repositories_url {
+                let installation_client = octocrab.installation(installation.id);
+
+                // Load repositories of this installation
+                let repos: InstallationRepositories = installation_client
+                    .get::<_, _, ()>(repositories_url, None)
+                    .await?;
+                for repo in repos.repositories {
+                    let Some(owner) = repo.owner else { continue; };
+                    let key = GitHubRepositoryKey::new(&owner.login, &repo.name);
+                    let access = Repository {
+                        client: installation_client.clone(),
```

Comment:
Since I had to look it up, octocrab refreshes expired tokens automatically under the hood.

Reply:
It does indeed, that is why I also cache these clients. I will add a comment to make it obvious.